### ISSUE TITLE
Eliminate all transitive guava-jdk5 dependencies

### ISSUE THREAD

**Issue by jbaldassari** [googleapis/google-cloud-java#2918]
Google Cloud dependencies are pulling in multiple incompatible versions of Guava.  Most depend on newer versions of Guava like ```com.google.guava:guava:20```, but others are pulling in ```com.google.guava:guava-jdk5:17.0```.  When I miss excluding that ancient jdk5 version from somewhere I end up with a bunch of classloader issues like:

```
java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkArgument(ZLjava/lang/String;J)V
```

For some reason I see this mostly with Cloud KMS (```com.google.apis:google-api-services-cloudkms```), but it actually appears to be coming from ```com.google.api-client:google-api-client:1.23.0```.  See http://mvnrepository.com/artifact/com.google.api-client/google-api-client/1.23.0 for example.

I'd be happy to contribute a PR myself, but I can't seem to find the right repo.

**Comment by garrettjonesgoogle** [comment]
This has been done in code (https://github.com/google/google-api-java-client/commit/1b35470936c83622167733b0c010a8321b03e3ab) but hasn't been released yet from google-api-java-client.

**Comment by jbaldassari** [comment]
Great to hear!  Thanks for the quick reply.

**Comment by pongad** [comment]
I'll add the "blocked" status since we're waiting for release.

**Comment by chingor13** [comment]
The google-api-java-client fix is out in 1.24.1.
